### PR TITLE
Allow overriding the VideoViewApi implementation

### PR DESCRIPTION
**Pull request by eygraber**

### CODE REVIEW

File: library/src/main/java/com/devbrackets/android/exomedia/ui/widget/EMVideoView.java [hunk 1]
```diff
@@ -204,11 +206,12 @@ private void readAttributes(Context context, @Nullable AttributeSet attrs) {
...
-    private void initView(Context context) {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN && EMDeviceUtil.isDeviceCTSCompliant()) {
-            View.inflate(context, R.layout.exomedia_exo_view_layout, this);
-        } else {
+    private void initView(Context context, @Nullable AttributeSet attrs) {
+        final boolean useLegacy = Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN || !EMDeviceUtil.isDeviceCTSCompliant();
+        if (useLegacy) {
             View.inflate(context, R.layout.exomedia_video_view_layout, this);
+        } else {
+            View.inflate(context, R.layout.exomedia_exo_view_layout, this);
```
Comment (brianwernick) [inline library/src/main/java/com/devbrackets/android/exomedia/ui/widget/EMVideoView.java]: If we are using ViewStub then we don't need separate layouts for legacy vs exo.

Comment (eygraber) [inline library/src/main/java/com/devbrackets/android/exomedia/ui/widget/EMVideoView.java]: Good point. I'll update that.

**Comment by eygraber** [comment]
Since there isn't any check to make sure that the `VideoViewApi` used for `EMVideoView_videoViewApiImpl` uses an `EMExoPlayer` and the one for `EMVideoView_videoViewApiImplLegacy` does not, I would suggest a warning (maybe in the README?) to only use them if you know what you are doing. Let me know if you want me to write that up and where.

**Comment by brianwernick** [comment]
I agree that It does need to be documented, however I'm not sure where it should go.  So that can be part of the update documentation phase before the release

**Comment by brianwernick** [comment]
:+1:

**Comment by eygraber** [comment]
Awesome. Amazing library BTW, and this 3.0 rewrite is great. I hope to contribute more in the future.

**Pull request merged**